src/app/asset-page/asset-page.component.ts

```diff
@@ -463,16 +463,32 @@ export class AssetPage implements OnInit, OnDestroy {
         this.showAccessDeniedModal = false
         this.showServerErrorModal = false
 
+        // asset arg is an Error object
         if (asset && asset['error']) {
             let err = asset['error']
-            if (err.status === 403 || err.message == 'Unable to load metadata!') {
+            if (err.status === 403 || err.message === 'Unable to load metadata!') {
                 // here is where we make the "access denied" modal appear
                 if (!this.encryptedAccess) {
                     this.showAccessDeniedModal = true
                 } else {
                     console.error('Failed to load externally shared asset', err)
                     this.showServerErrorModal = true
                 }
+
+                // AIR-2439 Workaround:
+                // When browser visits /public asset and receives 404,
+                // check if /public route and redirect to #/asset/assetId
+              if (this.isBrowser && err.message === 'Unable to load metadata!') {
+
+                this.route.url.subscribe((url: any[]) => {
+                  let path = url[0].path
+
+                  if (path === 'public' || path === 'object') {
+                    this._router.navigate(['/asset', url[1].path])
```

Comment:
I don't quite understand the purpose of this redirect? I would expect /public to behave just like the /asset route once loaded client-side?